diff --git a/daemon-user/src/ctrl_anime.rs b/daemon-user/src/ctrl_anime.rs
index 7961dd5..a680439 100644
--- a/daemon-user/src/ctrl_anime.rs
+++ b/daemon-user/src/ctrl_anime.rs
@@ -16,6 +16,7 @@ use crate::config::ConfigAnime;
 use crate::error::Error;
 
 #[derive(Debug, Clone, Deserialize, Serialize, Type)]
+#[zvariant(signature = "dict")]
 pub struct Timer {
     type_of: TimeType,
     /// If time type is Timer then this is milliseonds, otherwise it is
@@ -24,7 +25,7 @@ pub struct Timer {
     /// Used only for `TimeType::Timer`, milliseonds to fade the image in for
     fade_in: Option<u64>,
     /// Used only for `TimeType::Timer`, milliseonds to fade the image out for
-    fade_out: Option<u64>,
+    fade_out: Option<u64>
 }
 
 impl From<Timer> for AnimTime {
diff --git a/daemon-user/src/zbus_anime.rs b/daemon-user/src/zbus_anime.rs
index d08c218..4df553b 100644
--- a/daemon-user/src/zbus_anime.rs
+++ b/daemon-user/src/zbus_anime.rs
@@ -23,7 +23,10 @@
 
 use zbus::dbus_proxy;
 
-#[dbus_proxy(interface = "org.asuslinux.Daemon")]
+#[dbus_proxy(
+    interface = "org.asuslinux.Daemon",
+    assume_defaults = true
+)]
 trait Daemon {
     /// InsertAsusGif method
     fn insert_asus_gif(
